stan/math/prim/prob/exp_mod_normal_lpdf.hpp

```diff
@@ -43,58 +66,59 @@ return_type_t<T_y, T_loc, T_scale, T_inv_scale> exp_mod_normal_lpdf(
+  const auto& erfc_calc = to_ref(erfc(inner_term));
 
-  scalar_seq_view<T_y> y_vec(y);
-  scalar_seq_view<T_loc> mu_vec(mu);
-  scalar_seq_view<T_scale> sigma_vec(sigma);
-  scalar_seq_view<T_inv_scale> lambda_vec(lambda);
   size_t N = max_size(y, mu, sigma, lambda);
+  T_partials_return logp(0.0);
+  if (include_summand<propto>::value) {
+    logp -= LOG_TWO * N;
+  }
+  if (include_summand<propto, T_inv_scale>::value) {
+    logp += sum(log(lambda_val)) * N / size(lambda);
+  }
+  const auto& log_erfc_calc = log(erfc_calc);
+  logp
+      += sum(lambda_val * (mu_minus_y + 0.5 * lambda_sigma_sq) + log_erfc_calc);
 
-  for (size_t n = 0; n < N; n++) {
-    const T_partials_return y_dbl = value_of(y_vec[n]);
-    const T_partials_return mu_dbl = value_of(mu_vec[n]);
-    const T_partials_return sigma_dbl = value_of(sigma_vec[n]);
-    const T_partials_return lambda_dbl = value_of(lambda_vec[n]);
-    const T_partials_return inv_sigma = inv(sigma_dbl);
-    const T_partials_return sigma_sq = square(sigma_dbl);
-    const T_partials_return lambda_sigma_sq = lambda_dbl * sigma_sq;
-    const T_partials_return mu_minus_y = mu_dbl - y_dbl;
-    const T_partials_return inner_term
-        = (mu_minus_y + lambda_sigma_sq) * INV_SQRT_TWO * inv_sigma;
-    const T_partials_return erfc_calc = erfc(inner_term);
-    const T_partials_return deriv_logerfc
-        = -SQRT_TWO_OVER_SQRT_PI * exp(-square(inner_term)) / erfc_calc;
-
-    if (include_summand<propto>::value) {
-      logp -= LOG_TWO;
-    }
-    if (include_summand<propto, T_inv_scale>::value) {
-      logp += log(lambda_dbl);
-    }
-    logp += lambda_dbl * (mu_minus_y + 0.5 * lambda_sigma_sq) + log(erfc_calc);
-
-    if (!is_constant_all<T_y>::value) {
-      ops_partials.edge1_.partials_[n]
-          -= lambda_dbl + deriv_logerfc * inv_sigma;
-    }
-    if (!is_constant_all<T_loc>::value) {
-      ops_partials.edge2_.partials_[n]
-          += lambda_dbl + deriv_logerfc * inv_sigma;
+  operands_and_partials<T_y_ref, T_mu_ref, T_sigma_ref, T_lambda_ref>
+      ops_partials(y_ref, mu_ref, sigma_ref, lambda_ref);
+  if (!is_constant_all<T_y, T_loc, T_scale, T_inv_scale>::value) {
+    const auto& exp_m_sq_inner_term = exp(-square(inner_term));
+    const auto& deriv_logerfc = to_ref_if<
+        !is_constant_all<T_y, T_loc>::value + !is_constant_all<T_scale>::value
+            + !is_constant_all<T_inv_scale>::value
+        >= 2>(-SQRT_TWO_OVER_SQRT_PI * exp_m_sq_inner_term / erfc_calc);
+    if (!is_constant_all<T_y, T_loc>::value) {
+      const auto& deriv = to_ref_if < !is_constant_all<T_y>::value
+                          && !is_constant_all<T_loc>::value
+                                 > (lambda_val + deriv_logerfc * inv_sigma);
+      if (!is_constant_all<T_y>::value) {
+        ops_partials.edge1_.partials_ = -deriv;
+      }
+      if (!is_constant_all<T_loc>::value) {
+        ops_partials.edge2_.partials_ = std::move(deriv);
```

Comment:
Does moving into these do anything? I thought the partials were allocated in arena: https://github.com/stan-dev/math/blob/develop/stan/math/rev/functor/operands_and_partials.hpp#L242

Reply:
Now they are. Although that was not the case at the time I was writing this :)